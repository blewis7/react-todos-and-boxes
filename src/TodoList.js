import React, {useState} from "react";
import Todo from "./Todo";
import NewTodoForm from "./NewTodoForm";

const TodoList = () => {
    const [items, setItems] = useState([]);

    const addItem = (item) => {
        setItems(items => [...items, item]);
    };

    const remove = id => {
        setItems(items => items.filter(item => item.id !== id));
    };

    return (
        <div>
            <NewTodoForm createTodo={addItem} />
            <ul>
                {items.map(item => <Todo key={item.id} id={item.id} item={item.item} handleRemove={remove} />)}
            </ul>
        </div>
    )
}

export default TodoList