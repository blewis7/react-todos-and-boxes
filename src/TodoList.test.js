import React from "react";
import { fireEvent, render } from "@testing-library/react";
import TodoList from "./TodoList";

function addTodo(todoList, item="Celebrate finding money by spending it"){
    const itemInput = todoList.getByLabelText("New To-Do Item:");
    fireEvent.change(itemInput, { target: { value: item}});
    const button = todoList.getByText("Add To-Do Item");
    fireEvent.click(button);
};

// smoke test
it("renders without crashing", function() {
  render(<TodoList />);
});

// snapshot test
it("matches snapshot", function() {
  const {asFragment} = render(<TodoList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a new todo", function() {
    const todoList = render(<TodoList />);

    expect(todoList.queryByText("X")).not.toBeInTheDocument();

    addTodo(todoList);

    expect(todoList.queryByText("X")).toBeInTheDocument();
    expect(todoList.getAllByDisplayValue("")).toHaveLength(1);
});

it("can remove a todo", function(){
    const todoList = render(<TodoList />);

    addTodo(todoList);
    expect(todoList.queryByText("X")).toBeInTheDocument();
    const removeButton = todoList.getByText("X");
    fireEvent.click(removeButton);
    expect(todoList.queryByText("X")).not.toBeInTheDocument();
})