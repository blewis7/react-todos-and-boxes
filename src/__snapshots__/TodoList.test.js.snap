// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <div>
          <label
            for="todo"
          >
            New To-Do Item: 
          </label>
          <input
            id="todo"
            name="todo"
            placeholder="To-Do Item"
            required=""
            type="text"
            value=""
          />
        </div>
        <button>
          Add To-Do Item
        </button>
      </form>
    </div>
    <ul />
  </div>
</DocumentFragment>
`;
